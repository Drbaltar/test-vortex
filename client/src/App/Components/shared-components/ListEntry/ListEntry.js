import React from 'react';

const ListEntry = (props) => {
    const { entryFields } = props;
    
    return (
        <div className='row'>
            {getPopulatedFields(entryFields)}
        </div>
    );
};

const getPopulatedFields = (entryFields) => {
    return entryFields.map((entryField, index) => 
        <div key={index} className='col-sm'>
            <label>{entryField.label}</label>
            <input readOnly={true} className="form-control" type="text" placeholder={entryField.field}></input>
        </div>
    );
};

export default ListEntry;