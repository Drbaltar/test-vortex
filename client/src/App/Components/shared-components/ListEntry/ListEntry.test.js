import React from 'react';
import { shallow } from 'enzyme';
import ListEntry from './ListEntry';

const exampleEntry = [
    {
        label: 'Test Label 1:',
        field: 'Test Field 1:'
    },
    {
        label: 'Test Label 2:',
        field: 'Test Field 2:'
    },
    {
        label: 'Test Label 3:',
        field: 'Test Field 3:'
    },
    {
        label: 'Test Label 4:',
        field: 'Test Field 4:'
    }
];

describe('ListEntry', () => {
    const wrapper = shallow(<ListEntry entryFields={exampleEntry}/>);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('creates the correct number of fields based on input label/fields', () => {
        expect(wrapper.find('.col-sm').length).toBe(exampleEntry.length);
    });

    it('populates all the fields with the correct information', () => {
        let fieldNodes = wrapper.find('.col-sm');

        expect(getFieldValues(fieldNodes)).toEqual(exampleEntry);
    });
});

const getFieldValues = (nodes) => {
    let fieldValues = [];

    for (let index = 0; index < exampleEntry.length; index++) {
        fieldValues.push(
            {
                label: nodes.at(index).find('label').text(),
                field: nodes.at(index).find('input').props().placeholder
            }
        );
    }

    return fieldValues;
};