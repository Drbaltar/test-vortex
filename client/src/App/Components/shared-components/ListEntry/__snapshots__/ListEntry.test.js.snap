// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListEntry renders correctly 1`] = `
<div
  className="btn btn-block btn-outline-dark p-3"
  onClick={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-sm"
      key="0"
    >
      <label>
        Test Label 1:
      </label>
      <input
        className="form-control"
        placeholder="Test Field 1:"
        readOnly={true}
        type="text"
      />
    </div>
    <div
      className="col-sm"
      key="1"
    >
      <label>
        Test Label 2:
      </label>
      <input
        className="form-control"
        placeholder="Test Field 2:"
        readOnly={true}
        type="text"
      />
    </div>
    <div
      className="col-sm"
      key="2"
    >
      <label>
        Test Label 3:
      </label>
      <input
        className="form-control"
        placeholder="Test Field 3:"
        readOnly={true}
        type="text"
      />
    </div>
    <div
      className="col-sm"
      key="3"
    >
      <label>
        Test Label 4:
      </label>
      <input
        className="form-control"
        placeholder="Test Field 4:"
        readOnly={true}
        type="text"
      />
    </div>
  </div>
</div>
`;
